geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java

```diff
@@ -933,6 +933,24 @@ public NioSslEngine handshakeSSLSocketChannel(SocketChannel socketChannel, SSLEn
       BufferPool bufferPool)
       throws IOException {
     engine.setUseClientMode(clientSocket);
+    if (!clientSocket) {
+      engine.setNeedClientAuth(sslConfig.isRequireAuth());
+    }
+
+    if (clientSocket) {
+      if (sslConfig.doEndpointIdentification()) {
```

Comment:
this whole "if/else" statement is also in this class in configureClientSSLSocket. Should you extract it into a private method so both places use the same code?

Reply:
Thanks @dschneider-pivotal I have refactored it to a method. Both `SSLSocket` and `SSLEngine` returns a copy of SSLParameters to I could not completely do it inside the method. This new method returns params which are re-set into socket or engine.